engine/src/cmd/nebula.cpp

```diff
@@ -9,15 +9,147 @@
 #include "gfx/mesh.h"
 #include "gfx/cockpit.h"
 #include "universe.h"
+#include "unit_generics.h"
 
 #undef BOOST_NO_CWCHAR
 
 using XMLSupport::EnumMap;
 using XMLSupport::Attribute;
 using XMLSupport::AttributeList;
 
+using namespace XMLSupport;
+using namespace NebulaXML;
+using namespace VSFileSystem;
+
 extern double interpolation_blend_factor;
-void GameNebula::SetFogState()
+
+FOGMODE NebulaXML::parse_fogmode( string val )
+{
+    if (val == "exp")
+        return FOG_EXP;
+    else if (val == "exp2")
+        return FOG_EXP2;
+    else if (val == "linear")
+        return FOG_LINEAR;
+    else
+        return FOG_OFF;
+}
+
+
+
+//WARNING : USED TO CALL a GameUnit constructor but now Nebula::Nebula calls a Unit one
+Nebula::Nebula( const char *unitfile, bool SubU, int faction, Flightgroup *fg, int fg_snumber ) :
+    GameUnit< NebulaGeneric > ( unitfile, SubU, faction, string( "" ), fg, fg_snumber )
+{
+    fogme     = true;
+    string fullpath( unitfile );
+    fullpath += ".nebula";
+
+    LoadXML( fullpath.c_str() );
+
+    fadeinvalue = 0;
+    lastfadein  = 0;
+}
+
+
+
+void Nebula::beginElement( void *Userdata, const XML_Char *name, const XML_Char **atts )
+{
+    ( (Nebula*) Userdata )->beginElem( std::string( name ), AttributeList( atts ) );
+}
+
+void Nebula::beginElem( const std::string &name, const AttributeList &atts )
```

Comment:
this probably should also be called `beginElement()` and allow for function overloading to do the work.